pouch.go

```diff
@@ -183,11 +183,22 @@ func (p *pouch) resolveFile(fc FileConfig) error {
 		return err
 	}
 
-	err = ioutil.WriteFile(fc.Path, []byte(content), mode)
+	file, err := os.OpenFile(fc.Path, os.O_WRONLY|os.O_TRUNC|os.O_CREATE, mode)
+	if err != nil {
+		return fmt.Errorf("couldn't open %s file to be written: %s", fc.Path, err)
+	}
+	defer file.Close()
+
+	bytesWritten, err := file.Write([]byte(content))
 	if err != nil {
 		return fmt.Errorf("couldn't write secret in '%s': %s", fc.Path, err)
 	}
 
+	// Ensure file contents have been committed to disk
+	file.Sync()
```

Comment:
I think that doing a sync is not needed to solve this.

Thinking about the problems seen in #12, if kubelet restarts because it sees changes in disk, then it is reading already the new content.

Reply:
If we want to avoid problems related to things like kubelet reading the file when `WriteFile` truncates it, or when it is half-written, they we'd have to create a new file, and move it to overwrite the old one. (But I don't know if we are suffering because of this, and this would be another story in any case).

Reply:
My thought of changing to f.Sync() function was to try to ensure that each file is persisted in disk before starting with the next one. Thereby, if following files (or processes launched after this update) needed that file on disk, they would have it available.

Reply:
Btw, `Sync` can return an error :D but I don't know if it worths to handle it because this would mean something really broken in the host, or the file being something weird.

Reply:
Reverted , using again `ioutil.Writefile()`